config/config.go

```
@@ -0,0 +1,81 @@
+package config
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"github.com/daedaleanai/dbt/log"
+	"github.com/daedaleanai/dbt/util"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	Mirror string
+}
+
+var environment map[string]string
+var config *Config
+
+const configFileName string = "config.yaml"
+
+func init() {
+	environment = make(map[string]string)
+	for _, v := range os.Environ() {
+		parts := strings.Split(v, "=")
+		if len(parts) == 2 {
+			key := parts[0]
+			value := parts[1]
+			environment[key] = value
+		}
+	}
+}
+
+func getDbtConfigDir() string {
+	dbtConfigDir, ok := environment["DBT_CONFIG_DIR"]
+	if ok {
+		return dbtConfigDir
+	}
+
+	xdgConfigDir, ok := environment["XDG_CONFIG_DIR"]
+	if ok {
+		return path.Join(xdgConfigDir, "dbt")
+	}
+
+	homeDir, ok := environment["HOME"]
+	if ok {
+		return path.Join(path.Join(homeDir, ".config"), "dbt")
+	}
+
+	return ""
```

return an error here instead of encoding this in an empty string